pkg/sdk/go/users.go

```diff
@@ -321,6 +321,25 @@ func (sdk mgSDK) ListUserThings(userID string, pm PageMetadata, token string) (T
 	return tp, nil
 }
 
+func (sdk mgSDK) SearchUsers(pm PageMetadata, token string) (UsersPage, errors.SDKError) {
```

Comment:
how does this method differe from the `Users` method. since both call the same repository method, `retrieve basic info` and both have the same function signature, it is redundant

Reply:
I don't think so. It is calling search endpoint which calls the `searchBasicInfo` method

Reply:
ok, but the methods `RetrieveAllBasicInfo` and `SearchBasicInfo` are functionally simillar why do they have to be distinct functions

Reply:
There is a slight difference on how we construct the query. Also the `RetrieveAllBasicInfo` is supposed to be remove later from the conversation we had with @arvindh123. Initially I was to rename it to `searchBasicInfo` but it was agreed to keep it as it is and create the method seperately `searchBasicInfo`

Reply:
It is not the same, but we should `RetrieveBasicInfo` in `ListClients` in case the user is not an admin. @nyagamunene